### PR TITLE
OG initial implementation

### PR DESCRIPTION
**Pull request by Sudip7**
#600: initial implementation. @Pomax @alanmoo please assist. I am yet to find a solution for dynamic content generation.

### CODE REVIEW

File: pages/entry.jsx [hunk 1]
```diff
@@ -66,22 +66,28 @@ export default React.createClass({
...
       docTitle = `${this.state.entry.title}`;
       justPostedByUserMessage = this.state.justPostedByUser ? (<h5 className="col-12 text-center">Thanks for submitting!</h5>) : null;
       content = <ProjectCard {...Utility.processEntryData(this.state.entry)} onDetailView={true} />;
+      description = `${this.state.entry.description}`;
     }
 
     if (this.state.noData) {
       docTitle = NO_ENTRY_TITLE;
       content = NO_ENTRY_BLOCK;
+      description = '';
```
Comment (Pomax) [inline pages/entry.jsx]: we probably want to declare a new `NO_ENTRY_OG_DESCRIPTION` that has a generating entry string so that when search engines/twitter/etc try to read in that value, it has something in there, rather than being an empty string.

File: pages/entry.jsx [hunk 2]
```diff
@@ -66,22 +66,28 @@ export default React.createClass({
...
     }
 
     if (docTitle) {
-      docTitle = <Helmet><title>{ docTitle }</title></Helmet>;
+      docTitle = <Helmet>
+      <meta property="og:title" content= "{ `${this.state.entry.title}` }" />
```
Comment (Pomax) [inline pages/entry.jsx]: this deserves a bit of indentation. Also note that `content= "{ `${this.state.entry.title}` }"` doubles up the templating several times. Instead, just use `content={docTitle}` without the quotes, and without the template string syntax, and using the title

File: pages/entry.jsx [hunk 3]
```diff
@@ -66,22 +66,28 @@ export default React.createClass({
...
     let docTitle;
+    let description;
     let justPostedByUserMessage = null;
     let content;
 
     if (this.state.dataLoaded) {
       docTitle = `${this.state.entry.title}`;
       justPostedByUserMessage = this.state.justPostedByUser ? (<h5 className="col-12 text-center">Thanks for submitting!</h5>) : null;
       content = <ProjectCard {...Utility.processEntryData(this.state.entry)} onDetailView={true} />;
+      description = `${this.state.entry.description}`;
```
Comment (Pomax) [inline pages/entry.jsx]: There is no reason for template string syntax here: we're not templating anything into another string. Just use `description = this.state.entry.description;`

File: pages/entry.jsx [hunk 4]
```diff
@@ -66,22 +66,28 @@ export default React.createClass({
...
       docTitle = NO_ENTRY_TITLE;
       content = NO_ENTRY_BLOCK;
+      description = '';
     }
 
     if (docTitle) {
-      docTitle = <Helmet><title>{ docTitle }</title></Helmet>;
+      docTitle = <Helmet>
+      <meta property="og:title" content= "{ `${this.state.entry.title}` }" />
+      <meta property="og:description"  content="{ `${this.state.entry.description}` }" />
```
Comment (Pomax) [inline pages/entry.jsx]: Same here: just use `content={description}` (definitely don't use `this.state.entry.description`, we literally just invented the `description` variable to house the correct data =)

File: pages/entry.jsx [hunk 5]
```diff
@@ -66,22 +66,28 @@ export default React.createClass({
...
 
     if (this.state.noData) {
       docTitle = NO_ENTRY_TITLE;
       content = NO_ENTRY_BLOCK;
+      description = '';
     }
 
     if (docTitle) {
-      docTitle = <Helmet><title>{ docTitle }</title></Helmet>;
+      docTitle = <Helmet>
```
Comment (Pomax) [inline pages/entry.jsx]: why did you remove the `<title>{ docTitle }</title>` element?

**Comment by Pomax** [comment]
@Sudip7 were you still interested in this bug, or should it go back in the general bug pile?

**Comment by Sudip7** [comment]
@Pomax , m sorry was little occupied. Will update the PR today.

File: pages/entry.jsx [hunk 6]
```diff
@@ -5,7 +5,8 @@ import ProjectCard from '../components/project-card/project-card.jsx';
 import Service from '../js/service.js';
 import Utility from '../js/utility.js';
 
-const NO_ENTRY_TITLE = `Entry unavailable`;
+const NO_ENTRY_OG_TITLE = `Entry unavailable`;
```
Comment (Pomax) [inline pages/entry.jsx]: This has to stay `NO_ENTRY_TITLE` because this is data used for the page. We set the DOM's `<title>` element to this value. As such, rather than renaming it, keep it the old name, and instead just use that same value in the `meta property="og:title"` part (which you're already doing due to using `docTitle`). We don't need to put the "OG" part in the variable name, think of it instead as  putting "normal data" into opengraph meta tags as well. None of this data is strictly speaking "reserved for use by opengraph only".

**Comment by Pomax** [comment]
Thanks @Sudip7 - a small nit around naming variables, but if you change that we can start using this as a basis for building out opengraph data on all pages.

**Comment by Sudip7** [comment]
@Pomax should I need to rectify it and send a fresh pool request. There are some conflicts also.

**Comment by ryanwarsaw** [comment]
@Sudip7 I really encourage you to try and figure out how to resolve the conflicts without creating a new PR, if you need help I'm around on Mattermost all day so just let me know!

**Comment by xmatthewx** [comment]
@ryanwarsaw! hello, nice to see you around here.

**Comment by Sudip7** [comment]
tried to fix the conflict but somehow it still persists. Already spent 2 hours on this but no resolution. Allow me to create a new pull request. @Pomax .

 @ryanwarsaw thanks a lot for the inputs and assistance.

**Comment by Pomax** [comment]
@Sudip7 I've resolved the conflicts for you, but the original PR comments are still applicable.

**Comment by Sudip7** [comment]
@Pomax, is there any issue still there? The PR request has not been merged for very long time, any specific reason for that?

**Comment by Pomax** [comment]
Hi @Sudip7, it looks like the only reason was that it fell off our planning board.

File: pages/entry.jsx [hunk 7]
```diff
@@ -5,7 +5,8 @@ import ProjectCardDetialed from '../components/project-card/project-card-detaile
...
+const NO_ENTRY_OG_DESCRIPTION = `Description for this entry is currently unavailable`;
```
Comment (Pomax) [inline pages/entry.jsx]: We should grab this value from the actual entry object in follow-up. I've filed #749 for this

**Comment by mmmavis** [comment]
Hey @Sudip7 , I just realized this PR is now stale. Do you still have interests in working on this?

**Comment by Sudip7** [comment]
@mmmavis, thanks for reaching out. I am always interested to collaborate and ready to learn. Please suggest, what's the process to come out of this? Will try to pull that off this weekend.

**Comment by mmmavis** [comment]
Hey @Sudip7 like what Pomax mentioned in https://github.com/mozilla/network-pulse/pull/647/files#r134000701 we shouldn't rename two vars `NO_ENTRY_TITLE` and `NO_ENTRY_DESCRIPTION`. Once you have reverted the changes to those 2 vars we should be ready to push this to staging and test using FB's online debugger.

**Comment by mmmavis** [comment]
I'm doing some housekeeping for this repo before the Xmas break. Unfortunately this PR has been stale for a while so I'm gonna close it.
